test/idl/consistency.js

```diff
@@ -141,26 +162,31 @@ describe('Web IDL consistency', () => {
           partials.push(dfn);
         } else if (dfn.type === 'includes') {
           includes.push(dfn);
+        } else if (dfn.name) {
+          dfns.push(dfn);
         } else {
-          assert(dfn.name, `definition with no name in ${spec}`);
-          assert(!dfns.has(dfn.name), `duplicate definition of ${dfn.type} ${dfn.name}`);
-          dfns.set(dfn.name, dfn);
+          assert.fail(`unknown definition in ${spec}: ${JSON.stringify(dfn)}`);
         }
       }
     }
   });
 
+  it('unique definitions', () => {
+    assert.equal(nameMap(dfns).size, dfns.length);
+  });
+
   it('inheritance', () => {
-    for (const dfn of dfns.values()) {
+    const map = nameMap(dfns);
+    for (const dfn of map.values()) {
       if (dfn.inheritance) {
-        const parent = dfns.get(dfn.inheritance);
+        const parent = map.get(dfn.inheritance);
         assert(parent, `${dfn.name} inherits from missing type: ${dfn.inheritance}`);
-        assert.strictEqual(dfn.type, parent.type, `${dfn.name} inherits from wrong type: ${parent.type}`);
+        assert.equal(dfn.type, parent.type, `${dfn.name} inherits from wrong type: ${parent.type}`);
       }
     }
   });
 
-  it('flattening partials/mixins', () => {
-    flatten(dfns, partials, includes);
+  it('merging in partials/mixins', () => {
```

Comment:
I suggest to add a note for the future that the test should remain last because `mergeMembers` modifies the definitions in place, which could potentially afftect tests that we may add later on. Something like:

```suggestion
  // This test should remain the last one as it slightly modifies objects in dfns in place.
  it('merging in partials/mixins', () => {
```